time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalTimeArbitrary.java

```diff
@@ -0,0 +1,309 @@
+package net.jqwik.time.internal.properties.arbitraries;
+
+import java.time.*;
+import java.time.temporal.*;
+
+import org.apiguardian.api.*;
+
+import net.jqwik.api.*;
+import net.jqwik.api.arbitraries.*;
+import net.jqwik.time.api.arbitraries.*;
+
+import static java.time.temporal.ChronoUnit.*;
+import static org.apiguardian.api.API.Status.*;
+
+@API(status = INTERNAL)
+public class DefaultLocalTimeArbitrary extends ArbitraryDecorator<LocalTime> implements LocalTimeArbitrary {
+
+	private LocalTime timeMin = LocalTime.MIN;
+	private LocalTime timeMax = LocalTime.MAX;
+
+	private int hourMin = 0;
+	private int hourMax = 23;
+
+	private int minuteMin = 0;
+	private int minuteMax = 59;
+
+	private int secondMin = 0;
+	private int secondMax = 59;
+
+	private ChronoUnit precision = MILLIS;
+
+	@Override
+	protected Arbitrary<LocalTime> arbitrary() {
+
+		LocalTime effectiveMin;
+		try {
+			effectiveMin = calculateEffectiveMin();
+		} catch (NoValuesCanBeGeneratedException e) {
```

Comment:
What's the point of NoValuesCanBeGeneratedException? maybe you can use an IllegalArgumentException at the origin of the throws?

Reply:
Is an IllegalArgumentException okay at the runtime?

Reply:
It's made for that purpose and being thrown throughout jqwik.